File: 01-intro/sum-solved.java
int[][] arr = {{1, 2}, {1, 2}, {3, 4, {1, 2, 3}}};

public static int sum(int[][] arr) {
  int sum = 0;
  for (int[] row : arr) {
    for (int num : row) {
      sum += num;
    }
  }
  return sum;
}

File: 01-intro/sum-solved.ts
function sumArray(arr: any[]): number {
  let sum = 0;
  for (let i = 0; i < arr.length; i++) {
    if (Array.isArray(arr[i])) {
      sum += sumArray(arr[i]);
    } else {
      sum += arr[i];
    }
  }
  return sum;
}

const arr = [1, 2, [1, 2], [3, 4, [1, 2, 3]]];

const result = sumArray(arr);
console.log(result); // Output: 17
